t/ui/10-tests_overview.t

```diff
@@ -204,17 +205,55 @@ subtest 'filtering by architecture' => sub {
     };
 };
 
+subtest 'filtering by flavor' => sub {
+    $driver->get('/tests/overview?distri=opensuse&version=13.1&build=0091');
+
+    subtest 'by default, all flavors present' => sub {
+        check_build_0091_defaults;
+    };
+
+    subtest 'filter for specific flavors' => sub {
+        $driver->find_element('#filter-panel .card-header')->click();
+        $driver->find_element('#filter-flavor')->send_keys('DVD');
+        $driver->find_element('#filter-panel .btn-default')->click();
+
+        element_visible('#flavor_DVD_arch_i586',   qr/i586/);
+        element_visible('#flavor_DVD_arch_x86_64', qr/x86_64/);
+        element_not_present('#flavor_GNOME-Live_arch_i686');
+        element_not_present('#flavor_NET_arch_x86_64');
+    };
+};
+
 subtest 'filtering by test' => sub {
-    $driver->get('/tests/overview?test=textmode');
-
-    my @rows = $driver->find_elements('#content tbody tr');
-    is(scalar @rows, 1, 'exactly one row present');
-    like($rows[0]->get_text(), qr/textmode/, 'test is textmode');
-    like(
-        OpenQA::Test::Case::trim_whitespace($driver->find_element('#summary .card-header')->get_text()),
-        qr/Overall Summary of opensuse 13\.1 build 0092/,
-        'summary states "opensuse 13.1" although no explicit search params',
-    );
+
+    subtest 'request for specific test' => sub {
+        $driver->get('/tests/overview?test=textmode');
+
+        my @rows = $driver->find_elements('#content tbody tr');
+        is(scalar @rows, 1, 'exactly one row present');
+        like($rows[0]->get_text(), qr/textmode/, 'test is textmode');
+        like(
+            OpenQA::Test::Case::trim_whitespace($driver->find_element('#summary .card-header')->get_text()),
+            qr/Overall Summary of opensuse 13\.1 build 0092/,
+            'summary states "opensuse 13.1" although no explicit search params',
+        );
+    };
+
+    $driver->get('/tests/overview?distri=opensuse&version=13.1&build=0091');
+
+    subtest 'by default, all tests present' => sub {
+        check_build_0091_defaults;
+    };
+
+    subtest 'filter for specific test' => sub {
+        $driver->find_element('#filter-panel .card-header')->click();
+        $driver->find_element('#filter-test')->send_keys('textmode');
+        $driver->find_element('#filter-panel .btn-default')->click();
+
+        my @rows = $driver->find_elements('#content tbody tr');
+        is(scalar @rows, 1, 'exactly one row present');
+        like($rows[0]->get_text(), qr/textmode/, 'test is textmode');
```

Comment:
This leads to Perl errors if `is` fails and `@rows` is empty iirc. `find_element('#content tbody tr:only-child')` might be better?

Reply:
Honestly, i just copied it from another test. 😉